src/lib/selector/ProductSelector.ts

```diff
@@ -125,15 +125,18 @@ export class ProductSelector {
 
     public async getResourceJunctionObjects(connection: Connection, productIds: Array<String>) {
         //todo: add the productResource to schema class
+      try{
         const productResourcequery = "SELECT enxCPQ__Resource__r.enxCPQ__TECH_External_Id__c, enxCPQ__Criteria__c, enxCPQ__Applicable_Solution_Variants__c, enxCPQ__Product__r.enxCPQ__TECH_External_Id__c, CurrencyIsoCode, enxCPQ__Product__c, enxCPQ__TECH_External_Id__c, enxCPQ__Resource__c \
                                         FROM enxCPQ__ProductResource__c\
                                        WHERE enxCPQ__Product__r.enxCPQ__TECH_External_Id__c IN ('" + productIds.join('\',\'') + "')\
                                           OR enxCPQ__Product__r.RecordType.name = 'option'";
 
         const productResources = await Query.executeQuery(connection, productResourcequery, 'productResource');
-         
         return productResources;
-
+      } catch (e){
+        //if this falls it means that we retrieve from package version less than 9 and we have to skip querying productResources
+        console.log('-- Querying productResources omitted due to package version');
+      }
```

Comment:
Zastanawiam się, czy nie zrobić tu w przyszłości jakiegoś custom exception, bo teraz ten catch łapie nam wszystkie wyjątki, więc jak user źle skonfiguruje fieldsToIgnore, na przykład, to może mu mylnie polecieć ten wyjątek. Ale tak czy siak spoko opcja z catchem.

Reply:
Trafna uwaga, nie wiem tylko czy tworzenie custom exception jest najlepszym pomysłem akurat w tej sytuacji bo rzucając exception zatrzymamy egzekucje kodu.  Moge zabezpieczyć sytuacje gdy ktoś poda złe pole sprawdzając errorCode jesli jest inny niż INVALID_TYPE wtedy rzucimy exception.